Classes/common/CDTIncrementalStore.m

```diff
@@ -922,11 +932,14 @@ - (BOOL)insertManagedObject:(NSManagedObject *)mo error:(NSError **)error
  *  Update existing object in the database
  *
  *  @param mo    Managed Object
+ *  @param changes Dictionary of [key:NSString, value:id] of changes to make to mo contents
  *  @param error Error
  *
  *  @return YES/NO
  */
-- (BOOL)updateManagedObject:(NSManagedObject *)mo error:(NSError **)error
+- (BOOL)updateManagedObject:(NSManagedObject *)mo
+                withChanges:(NSDictionary *)changes
```

Comment:
Sadly, this split is incorrect.
The fact that you have an `NSManagedObject` means that you used the context, which means you fetched.
The flow for batch processing should never touch the context that was passed in to you.
You should be passing in a `NSManagedObjectID` or a `docID` and the `changes` should be the `props` that is built here.